backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/attachments/AttachmentReferenceRepository.java

```diff
@@ -0,0 +1,184 @@
+import org.adempiere.ad.dao.ICompositeQueryFilter;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.dao.IQueryBuilder;
+import org.adempiere.util.lang.impl.TableRecordReference;
+import org.compiere.model.I_AD_Attachment_MultiRef;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+
+import static org.adempiere.model.InterfaceWrapperHelper.load;
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+
+@RequiredArgsConstructor
+@Repository
+class AttachmentReferenceRepository
+{
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+
+	public AttachmentReference getById(@NonNull final AttachmentReferenceId id)
+	{
+		final I_AD_Attachment_MultiRef attachmentEntryRecord = load(id, I_AD_Attachment_MultiRef.class);
+		return forRecord(attachmentEntryRecord);
+	}
+
+	public ImmutableList<AttachmentReference> getByReferencedRecord(@NonNull final TableRecordReference referencedRecord)
+	{
+		return getByReferencedRecords(ImmutableList.of(referencedRecord));
+	}
+
+	public ImmutableList<AttachmentReference> getByReferencedRecords(@NonNull final Collection<TableRecordReference> referencedRecords)
+	{
+
+		final IQueryBuilder<I_AD_Attachment_MultiRef> queryBuilder = queryBL.createQueryBuilder(I_AD_Attachment_MultiRef.class)
+				.setJoinOr();
+		for (final TableRecordReference referencedRecord : referencedRecords)
+		{
+			final ICompositeQueryFilter<I_AD_Attachment_MultiRef> filter = queryBL
+					.createCompositeQueryFilter(I_AD_Attachment_MultiRef.class)
+					.addOnlyActiveRecordsFilter()
+					.addEqualsFilter(I_AD_Attachment_MultiRef.COLUMN_Record_ID, referencedRecord.getRecord_ID())
+					.addEqualsFilter(I_AD_Attachment_MultiRef.COLUMNNAME_AD_Table_ID, referencedRecord.getAD_Table_ID());
+
+			queryBuilder.filter(filter);
+		}
+
+		return queryBuilder.create()
+				.stream()
+				.map(this::forRecord)
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	private AttachmentReference forRecord(@NonNull final I_AD_Attachment_MultiRef multiRefRecord)
+	{
+		final AttachmentEntryId attachmentEntryId = AttachmentEntryId.ofRepoId(multiRefRecord.getAD_AttachmentEntry_ID());
+		final AttachmentReferenceId id = AttachmentReferenceId.ofRepoIdOrNull(multiRefRecord.getAD_Attachment_MultiRef_ID());
+
+		return AttachmentReference.builder()
+				.id(id)
+				.attachmentEntryId(attachmentEntryId)
+				.recordRef(TableRecordReference.of(multiRefRecord.getAD_Table_ID(), multiRefRecord.getRecord_ID()))
+				.attachmentNameOverride(multiRefRecord.getFileName_Override())
+				.build();
+	}
+
+	public ImmutableList<AttachmentReference> saveAll(@NonNull final Collection<AttachmentReference> attachmentReferences)
+	{
+		return attachmentReferences.stream()
+				.map(this::save)
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	public AttachmentReference save(@NonNull final AttachmentReference attachmentReference)
```

Comment:
for the impl o this method IMHO u shall just call saveAll(ImmutableList.o(attahcmentReference));

Reply:
i think while your reviewed yesterday, i made changes. now i think the comment is outdated